### PR TITLE
Enable tls only

### LINKED ISSUE #6: SSL support
Currently everything happens over plain TCP but since twitch supports SSL, we should too.

https://dev.twitch.tv/docs/irc#connecting-to-twitch-irc

### PR DESCRIPTION
**Pull request by VitorFalcao**
Fix #6 
- Ditched plain TCP and implemented TLS.
- Updated tests to create local servers with TLS.
- Edited the Makefile to create a private key and a certificate for testing. It removes them after the test is done.
- Added a check to "Connect()" in "client.go" regarding the server address. If it is localhost, we configure the TLS client to accept any certificate. If not, it checks the server's certificate chain and hostname.

### CODE REVIEW

**Comment by VitorFalcao** [comment]
From I've seen, the Travis CI build fails because it does not create the localhost server certificate, which is done by the Makefile. How should I fix this?

**Comment by VitorFalcao** [comment]
It seems to me the build is failling because it is creating the server certificate. It works locally... Probably an issue regarding the openssl command. Any ideias?

**Comment by gempir** [comment]
I think for tests it would be ok to keep the cert and key, we don't want all the overhead from creating them again and again each test. Should fix travis issues aswell

File: client.go [hunk 1]
```diff
@@ -129,7 +138,8 @@ func (c *Client) send(line string) {
 			time.Sleep(time.Millisecond * 2)
 			continue
 		}
-		fmt.Fprint(*c.connection, line+"\r\n")
+		//fmt.Fprint(*c.connection, line+"\r\n")
```
Comment (gempir) [inline client.go]: Comments should be removed. I prefer the direct way of writing, instead of using fmt. That's a good change.

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,9 @@
 test:
```
Comment (gempir) [inline Makefile]: Like I said, probably best to just store the cert and key in the repo, we don't need to generate them everytime

**Comment by gempir** [comment]
@VitorFalcao Thank you for your contribution! Great work!

**Comment by VitorFalcao** [comment]
Thank you =D This was my first contribution ever. Really liked doing it!

**Pull request merged**